BasicAPI6/Notebook.java

```diff
@@ -0,0 +1,168 @@
+import java.io.IOException;
+// описание свойств ноутбука и методов работы с его свойствами
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+
+public class Notebook {
+    private HashMap<String, Float> notebook;
+    // экземпляр класса Notebook содержит соответствующие ID (для связанных таблиц), физические характеристики ноутбука и его цену.
+    // поскольку часть характеристик имеет дробное значение, для хранения используется float,
+    // а для извлечения методы getInt и getFloat
+    public static final String[] KEYS = {"BrandID", "ModelID", "OperatingSystemID", "ScreenSize", "ScreenResolutionX", 
+    "ScreenResolutionY", "CPUFrequency", "CoresNumber", "RAMCapacity", "DiskCapacity", "ColorID", "Price" };//Поля записи и порядок их следования
+    public static final String[] KEYSDESCRIPTION = {"Название бренда", "Название модели", "Операционная система", "Размер экрана (дюймы)", "Разрешение по горизонтали",
+    "Разрешение по вертикали", "Частота процессора (ГГц)", "Число ядер", "Объем оперативной памяти (Гб)", "Объем накопителя (Гб)", "Цвет", "Цена"};
+    // ноутбук характеризуется огромным количеством параметров. Тут приведена лишь малая часть. Если заводить под каждый параметр свою переменную и 
+    // писать set и get методы для работы с каждой переменной, то работать с таким классом будет невозможно. Поэтому сделана имитация HashMap (доступ по ключу)
+    // В этом случае можно легко добавлять новые параметры меняя лишь поля записи (KEYS)
+    public static final List<String> IDKEYS = List.of("BrandID", "ModelID", "OperatingSystemID","ColorID");
+    // Ключи которые связаны с ID связанных таблиц
+    public static final float NUL = 0; // Число для обозначения отсутствия данных
+    public static final float ERROR = -1; // Число для обозначения ошибки
+    public static final String FIELDSSEPARATOR = ";"; // Разделитель полей записи в строковом представлении
+    public static final String KEYSEPARATOR = ":"; // Разделитель пары ключ - значение 
+
+    public static Set<Notebook> loadSet(String fileName){
+        Set<Notebook> notebooks = new HashSet<>();
+        //загрузка из файла в множество ноутбуков
+        try{
+            File file = new File(fileName);
+            Scanner input = new Scanner(file);
+            String textLine;
+            while (input.hasNext()){
+                textLine = input.nextLine();
+                if (!textLine.isEmpty()){
+                    Notebook notebook = new Notebook(textLine);
+                    //Экземпляр Notebook создается из строкового представления и добавляется в множество
+                    notebooks.add(notebook);
+                }
+            }
+        input.close();
+        } catch (IOException e){
+            System.out.println("Ошибка чтения файла " + fileName);
+        }
+        return notebooks;
+    }
+    
+    Notebook(){
+        this.notebook = new HashMap<>();
```

Comment:
Better to use explicit verified fields like
private final int screenResolution
private final String model ... etc
instead of proxying hashmap calls.
So you always know that data is valid

Reply:
The idea is that Notebook object contains too many fields so it is easier to work with it in Map manner getting data through the keys. Notebook class controls the keys so user cannot add/get data by a non-existent key.